### ISSUE TITLE
[Question] esbuild fails to replace "path" in libsodium.js

### ISSUE THREAD

**Issue by LukasForst** [evanw/esbuild#1786]
I'm experimenting with esbuild, but while trying to build an application for brwoser that imports `libsodium-wrappers-sumo` package, esbuild fails to package it with following error:
```
$ node build.js
 > node_modules/libsodium-sumo/dist/modules-sumo/libsodium-sumo.js:1:1090: error: Could not resolve "path" (use "platform: 'node'" when building for node)
    1 │ ...de,Q=!n&&!e&&!o;var C,c,i,_,s,y="";function E(A){return r.locateFile?r.locateFile(A,y):y+A}e?(y=o?require("path").dirname(y)+"/":__dirname+"/",C=function(A,g){var I=HA(A);return I?g?I:I.toString():(_||(_=require("fs"))...
```
I'd expect that it will replace `path` as instructed in package.json according to the documentation.

`package.json` looks like that (with `browser` set as in [the doc](https://esbuild.github.io/getting-started/#bundling-for-the-browser)).
```json
{
  "name": "esbuild",
  "version": "1.0.0",
  "main": "build.js",
  "license": "MIT",
  "dependencies": {
    "esbuild": "^0.13.14",
    "libsodium-wrappers-sumo": "^0.7.9"
  },
  "scripts": {
    "build": "node build.js"
  },
  "browser": {
    "path": "path-browserify"
  }
}

```
`app.jsx` then like that
```js
import * as sodium from 'libsodium-wrappers-sumo';

console.log(sodium.base64_variants)
```

and finally `build.js`:
```js
require('esbuild').build({
  entryPoints: ['app.jsx'],
  bundle: true,
  platform: 'browser',
  outfile: 'out.js',
  external: ['fsevents'],
}).catch(() => process.exit(1))
```

Any idea how to force esbuild to replace it for the browser package?

**Comment by hyrious** [comment]
Did you install `path-browserify`? Otherwise esbuild won't find it.
In fact, the `browser` field to esbuild is just a handy way to define alias.

**Comment by LukasForst** [comment]
@hyrious yup, adding `"path-browserify": "^1.0.1"` to `package.json` and installing it does not change a thing

**Comment by evanw** [comment]
This is unfortunately how the `browser` field works. I didn't design this behavior and I agree it's unfortunate. The `browser` field apparently originated with [Browserify](https://browserify.org/). As far as I can tell, Browserify basically only respects the `browser` field within a package, not across packages. This is determined by a big support matrix I made to try to reverse-engineer what the behavior should be, since there's no actual specification for how `browser` should behave: https://github.com/evanw/package-json-browser-tests. So the `browser` field is really only for package authors, not package consumers.

A way to do this substitution that works across packages is to write a quick esbuild plugin to do it. Documentation for the plugin API is here: https://esbuild.github.io/plugins/#resolve-callbacks.

Closing because it looks like esbuild's interpretation of the `browser` field is correct here.
